### PR TITLE
[GPII-4015]: Use K8s 1.13.7-gke.8

### PR DESCRIPTION
**Pull request by natarajaya**
After adding ServiceEntries to allow CloudTrace traffic from pods in #429, I was still seeing Flowmanager egress issues.
Adding VirtualServices did not seem to help, so I decided to try to upgrade to K8s `1.13.7-gke.8` that uses Istio `1.1.7`, it worked – I am unable to reproduce Flowmanager Locust issues anymore.

### CODE REVIEW

**Comment by stepanstipl** [comment]
Thanks @natarajaya 

> After adding ServiceEntries to allow CloudTrace traffic from pods in #429, I was still seeing Flowmanager egress issues.

Can you please commit the code incl. VirtualServices? I don't see any added in the #429 PR, and I would like to have a look before we decide to upgrade to 1.13 as we will lose some features like consistency between versions of ephemeral and non-ephemetal clusters after creation.

Also note that this will not upgrade nodes immediately on non-ephemeral clusters. Is upgrading only masters sufficient to resolve the CloudTrace/Flowmanager egress issues?

**Comment by natarajaya** [comment]
> Can you please commit the code incl. VirtualServices?

Done.

> we will lose some features like consistency between versions of ephemeral and non-ephemetal clusters after creation

But thats only for a while until 1.13 becomes GKE-default.

> Is upgrading only masters sufficient to resolve the CloudTrace/Flowmanager egress issues?

Nope, workers needs to be upgraded as well.

**Comment by stepanstipl** [comment]
Thanks @natarajaya

> But thats only for a while until 1.13 becomes GKE-default.

Yep, but this has other side effects - like cluster upgrade right after cluster creation (when we run locust tests) and inconsistency between master & nodes in ephemeral envs.,  which were previously giving us issues.

> Nope, workers needs to be upgraded as well.

Then it should probably be mentioned as part of this PR to trigger the node upgrade.

Anyway, I agree with the proposal in #429 and put this on hold till we see what's the situation after that PR is merged.

**Comment by natarajaya** [comment]
> Yep, but this has other side effects - like cluster upgrade right after cluster creation (when we run locust tests) and inconsistency between master & nodes in ephemeral envs., which were previously giving us issues

@stepanstipl Interesting. I have not noticed any of that while testing this with my dev cluster. Let's then hope that VirtualService fixes from #429 work.

**Comment by natarajaya** [comment]
> Then it should probably be mentioned as part of this PR to trigger the node upgrade.

Yeah, I have not worked on deployment plan for this yet.

File: gcp/modules/gke-cluster/main.tf [hunk 1]
```diff
@@ -23,6 +23,12 @@ variable "node_count" {
...
 
+# When this variable is blank, GKE-default K8s version is used.
+# Check data.gke_version_assert for more details.
+variable "kubernetes_version" {
+  default = "1.13.7-gke.8"
+}
+
 variable "expected_gke_version_prefix" {
   default = "1.12"
 }
```
Comment (mrtyler) [inline gcp/modules/gke-cluster/main.tf]: I find the interaction of `kubernetes_version` and `expected_gke_version_prefix` kind of confusing. Maybe change the latter to `expected_default_gke_version_prefix`?

**Comment by natarajaya** [comment]
> I feel a little lost in this PR, so apologies if I'm lacking context

@mrtyler This PR is our last resort against Istio's TLS routing issues described in #429. But after VirtualService fixes, we don't see them anymore (at least in CI, locally I am bumping into them at least 1-2 times a week). So, hopefully, we can abandon this and live with GKE-default K8s.

I propose to give it one more week and then I'll close this if situation remains stable.

**Comment by stepanstipl** [comment]
> I propose to give it one more week and then I'll close this if situation remains stable.

Can we close this one? (alway can be reopened should we decide so)

**Pull request closed without merging**